src/base.rs

```diff
@@ -0,0 +1,290 @@
+//! Primitives for object and array transmutation.
+//!
+//! The functions in this module are very unsafe and their use is not
+//! recommended unless you _really_ know what you are doing.
+use crate::error::Error;
+use crate::guard::{Guard, SingleManyGuard, SingleValueGuard, PermissiveGuard};
+#[cfg(feature = "std")]
+use core::mem::forget;
+use core::mem::size_of;
+use core::slice;
+
+/// Convert a byte slice into a single instance of a `Copy`able type.
+///
+/// The byte slice must have at least enough bytes to fill a single instance of
+/// a type, extraneous data is ignored.
+///
+/// # Safety
+///
+/// - This function does not perform memory alignment checks. The beginning of
+/// the slice data must be properly aligned for accessing the value of type `T`.
+/// - The byte data needs to correspond to a valid `T` value.
+///
+/// Failure to fulfill any of the requirements above is dangerous and may eat
+/// your laundry.
+///
+/// # Errors
+///
+/// An error is raised if the slice does not have enough bytes for a single
+/// value `T`.
+///
+/// # Examples
+///
+/// ```
```

Comment:
Shouldn't this test be `no_run` in keeping with the convention in other places since `u8`s don't guarantee alignments?

Reply:
Great question! Should I apply this to all doctests with this issue?

Reply:
Well I mean, probably, unless we want spuriously failing tests, that is :Р

Reply:
I agree that we don't want that, and I'll take care of adding `no_run` where appropriate. But note that we haven't had spuriously failing tests so far, because we know that the real™ consequences of unaligned memory access on the architectures that we are testing on at the moment are not serious enough for the tests to fail. Ideally, we'd have a target triplet with some extra configurations that would force the program to seriously misbehave, so that we could track these buggers down. Unfortunately, none of us managed to do such a thing yet. With that said, this part right here deserves its own issue.

Reply:
Tremendous!